### PR TITLE
Added support for multiple TLS domains and example of multidomain.

### PR DESCRIPTION
**Pull request by dgrr**
I am demonstrating how much I love this package xd

### CODE REVIEW

File: examples/multidomain/multidomain.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+			h(ctx)
+		},
+	}
+
+	// preparing first host
+	cert, priv, err := GenerateCert("localhost")
+	if err != nil {
+		panic(err)
+	}
+	domains["localhost"] = func(ctx *fasthttp.RequestCtx) {
```
Comment (erikdubbelboer) [inline examples/multidomain/multidomain.go]: This should be `localhost:8080`.

File: server.go [hunk 2]
```diff
@@ -1223,48 +1231,87 @@ func (s *Server) ListenAndServeTLSEmbed(addr string, certData, keyData []byte) e
...
 //
 // certFile and keyFile are paths to TLS certificate and key files.
+//
+// If the certFile or keyFile has not been provided the server structure,
+// the function will use previously added TLS configuration.
 func (s *Server) ServeTLS(ln net.Listener, certFile, keyFile string) error {
-	lnTLS, err := newTLSListener(ln, certFile, keyFile)
-	if err != nil {
-		return err
+	s.AppendCert(certFile, keyFile)
```
Comment (erikdubbelboer) [inline server.go]: You need to check the error returned by this function. If someone now specified and invalid certificate they will get an `No certFile or keyFile has been provided` error which is very confusing and incorrect.

**Comment by dgrr** [comment]
Done. @erikdubbelboer

File: server.go [hunk 3]
```diff
@@ -1223,48 +1231,91 @@ func (s *Server) ListenAndServeTLSEmbed(addr string, certData, keyData []byte) e
 // ServeTLS serves HTTPS requests from the given listener.
 //
 // certFile and keyFile are paths to TLS certificate and key files.
+//
+// If the certFile or keyFile has not been provided the server structure,
+// the function will use previously added TLS configuration.
 func (s *Server) ServeTLS(ln net.Listener, certFile, keyFile string) error {
-	lnTLS, err := newTLSListener(ln, certFile, keyFile)
-	if err != nil {
+	if err := s.AppendCert(certFile, keyFile); err != nil {
```
Comment (erikdubbelboer) [inline server.go]: Now your example program doesn’t work anymore as you are calling this function with an empty certFile and keyFile. You should only call AppendCert here if they aren’t empty strings.

**Comment by dgrr** [comment]
I fixed the error and added a test.

**Comment by erikdubbelboer** [comment]
Thanks.

**Pull request merged**